### PR TITLE
Removed setTo argument from xPoweredBy

### PR DESCRIPTION
**Pull request by ameen-abdeen**
Removed XPoweredByOptions and make xPoweredBy take no arguments.
Removed the test
Updated the changelog and linked it to the given wiki page

### CODE REVIEW

File: middlewares/x-powered-by/index.ts [hunk 1]
```diff
@@ -1,25 +1,13 @@
...
+function getUpdaterFn(): (res: ServerResponse) => void {
+  return (res) => {
+    res.removeHeader("X-Powered-By");
+  };
 }
 
-function xPoweredBy(options: Readonly<XPoweredByOptions> = {}) {
-  const updateResponse = getUpdaterFn(options.setTo);
+function xPoweredBy() {
+  const updateResponse = getUpdaterFn();
```
Comment (EvanHahn) [inline middlewares/x-powered-by/index.ts:10]: We don't need a fancy updater function any more—we can just return the middleware. For example:

```ts
function xPoweredByMiddleware(
  _req: IncomingMessage,
  res: ServerResponse,
  next: () => void
) {
  res.removeHeader("X-Powered-By");
  next();
}

function xPoweredBy() {
  return xPoweredByMiddleware;
}
```

Could you update so that it looks like that?

File: middlewares/x-powered-by/CHANGELOG.md [hunk 2]
```diff
@@ -12,3 +12,9 @@
 - Excluded some files from npm package
 
 Changes in versions 1.0.0 and below can be found in [Helmet's changelog](https://github.com/helmetjs/helmet/blob/master/CHANGELOG.md).
+
+### Changed
```
Comment (EvanHahn) [inline middlewares/x-powered-by/CHANGELOG.md:16]: This should be at the top of the changelog under a new "Unreleased" heading. It should roughly have this shape:

```md
# Changelog

## Unreleased

### Changed

- Removed `setTo` option. ...

## 1.1.0 - 2019-05-26

...
```

File: middlewares/x-powered-by/CHANGELOG.md [hunk 3]
```diff
@@ -12,3 +12,9 @@
...
+
+- Removed setTo option from x-powered-by middleware. x-powered-by middleware no longer accepts any arguments.
+
+Changes in versions 4.0.0 and below can be found in [Helmet's changelog](https://github.com/helmetjs/helmet/blob/master/CHANGELOG.md).
```
Comment (EvanHahn) [inline middlewares/x-powered-by/CHANGELOG.md:20]: We can delete this line.

File: CHANGELOG.md [hunk 4]
```diff
@@ -8,6 +8,7 @@
 - `helmet.noSniff` is no longer a separate package. This should have no effect on end users.
 - `helmet.permittedCrossDomainPolicies` is no longer a separate package. This should have no effect on end users.
 - `helmet.referrerPolicy` is no longer a separate package. This should have no effect on end users.
+- `helmet.hidePoweredBy` no longer accepts arguments. Remove `setTo` option. See [#224](https://github.com/helmetjs/helmet/wiki/How-to-set-a-custom-X%E2%80%93Powered%E2%80%93By-header)
```
Comment (EvanHahn) [inline CHANGELOG.md:11]: This should link to #224. Could we change this bullet point to:

```markdown
- `helmet.hidePoweredBy` no longer accepts arguments. See [this article](https://github.com/helmetjs/helmet/wiki/How-to-set-a-custom-X%E2%80%93Powered%E2%80%93By-header) to see how to replicate the removed behavior. See [#224](https://github.com/helmetjs/helmet/issues/224).
```

File: middlewares/x-powered-by/CHANGELOG.md [hunk 5]
```diff
@@ -12,3 +12,9 @@
 - Excluded some files from npm package
 
 Changes in versions 1.0.0 and below can be found in [Helmet's changelog](https://github.com/helmetjs/helmet/blob/master/CHANGELOG.md).
+
+### Changed
+
+- Removed setTo option from x-powered-by middleware. x-powered-by middleware no longer accepts any arguments.
```
Comment (EvanHahn) [inline middlewares/x-powered-by/CHANGELOG.md:18]: Could we change this bullet point to:

```markdown
- Removed `setTo` option. See [this article](https://github.com/helmetjs/helmet/wiki/How-to-set-a-custom-X%E2%80%93Powered%E2%80%93By-header) to see how to replicate the removed behavior. See [#224](https://github.com/helmetjs/helmet/issues/224).

**Comment by ameen-abdeen** [comment]
I have resolved all the comments but there is a merge conflict in the change log which I couldn't find a way to resolve

**Comment by ameen-abdeen** [comment]
> Thanks for doing this! I left a few comments. Let me know if I can help further.

It is my pleasure to contribute to the world of opensource . There was a merge conflict in `middlewares/x-powered-by/CHANGELOG.md` which I have resolved but not sure whether it is right. There was `2.0.0 - Unreleased` tag. I was not quite sure whether my change was required under it

File: middlewares/expect-ct/index.ts [hunk 6]
```diff
@@ -21,12 +21,12 @@ function parseMaxAge(value: void | number): number {
 function getHeaderValueFromOptions(options: Readonly<ExpectCtOptions>): string {
   const directives: string[] = [];
 
+  directives.push(`max-age=${parseMaxAge(options.maxAge)}`);
```
Comment (EvanHahn) [inline middlewares/expect-ct/index.ts:24]: This shouldn't be done as part of this pull request. Could you revert? We should also revert the changes in `test/expect-ct.test.ts`.

(It can also be done against the `master` branch instead of the `4.x` branch.)

File: package.json [hunk 7]
```diff
@@ -69,9 +69,9 @@
     "pretest": "npm run lint",
     "prepublishOnly": "npm run build",
     "lint": "npm run lint:eslint && npm run lint:prettier",
-    "lint:eslint": "eslint '**/*.ts'",
-    "lint:prettier": "prettier --check '**/*{md,js,json,ts}'",
-    "format": "prettier --write '**/*{md,js,json,ts}'",
+    "lint:eslint": "eslint **/*.ts",
```
Comment (EvanHahn) [inline package.json:72]: I don't think this file should've changed. Was it not working for you?

File: middlewares/x-powered-by/CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,11 @@
 # Changelog
 
+## Unreleased
```
Comment (EvanHahn) [inline middlewares/x-powered-by/CHANGELOG.md:3]: This looks good. Could you move the bullet point under the "Removed" header?

File: package.json (hunk 7 above)
Comment (ameen-abdeen) [reply package.json:72]: It was not working for me.

Comment (EvanHahn) [reply package.json:72]: Hmm, okay. That's something we should fix, but not in this pull request. I created #227 to note this, and I'll do it today.

File: middlewares/x-powered-by/CHANGELOG.md (hunk 8 above)
Comment (ameen-abdeen) [reply middlewares/x-powered-by/CHANGELOG.md:3]: Just the bullet point or the entire section

File: package.json (hunk 7 above)
Comment (EvanHahn) [reply package.json]: Scripts should now work.

File: middlewares/x-powered-by/CHANGELOG.md (hunk 8 above)
Comment (EvanHahn) [reply middlewares/x-powered-by/CHANGELOG.md:3]: Just the bullet point.

**Comment by EvanHahn** [comment]
Thanks for your help here! I will take a look tomorrow.

**Comment by ameen-abdeen** [comment]
> Looks good! Thank you!!
> 
> Could you fix the merge conflicts? Once you do that, I'll merge. Let me know if you need help with that.
> 
> ~I'm going to merge this now.~ Feel free to add yourself to the list of `contributors` in `package.json` in a separate pull request.

I don't see any merge conflicts between `helmetjs:4.x` and `ameen-abdeen:master`.

**Comment by EvanHahn** [comment]
My mistake. Just merged this! Thanks so much!

Would you like me to add you to the `contributors` in `package.json`? If so, what name, email, and website should I use? (Email and website are optional.)

**Comment by ameen-abdeen** [comment]
> My mistake. Just merged this! Thanks so much!
> 
> Would you like me to add you to the `contributors` in `package.json`? If so, what name, email, and website should I use? (Email and website are optional.)

That would be great. Name: Ameen Abdeen Email: ameen.abdeen.se@gmail.com. I don't have a website yet.

**Comment by EvanHahn** [comment]
Credited you in e1746c15f475bcccbc6fe3f2c674cebbe5892395! Thanks again.
